app/services/s3_query_service.rb

```diff
@@ -75,14 +75,19 @@ def client
   # For docs see:
   # * https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Client.html#list_objects_v2-instance_method
   # * https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Client.html#get_object_attributes-instance_method
-  # @return [<S3File>] An Array of S3File objects
+  # @return Hash with two properties {objects: [<S3File>], ok: Bool}
+  #   objects is an Array of S3File objects
+  #   ok is false if there is an error connecting to S3. Otherwise true.
   def data_profile
     objects = []
     resp = client.list_objects_v2({ bucket: bucket_name, max_keys: 1000, prefix: prefix })
     resp.to_h[:contents]&.each do |object|
       s3_file = S3File.new(filename: object[:key], last_modified: object[:last_modified], size: object[:size])
       objects << s3_file
     end
-    objects
+    { objects: objects, ok: true }
+  rescue => ex
+    Rails.logger.error("Error querying S3. Bucket: #{bucket_name}. Prefix: #{prefix}. Exception: #{ex.message}")
```

Comment:
Log details about the error here and return false to the caller (in this case the controller) so that we can handle the display as well without having to deal with the details (e.g the bucket_name) at that level.